internal/serve/sep24frontend/app/src/pages/Success.tsx

```diff
@@ -1,14 +1,26 @@
-import { FC } from "react";
+import { FC, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
 import { Trans, useTranslation } from "react-i18next";
 import { Button, Heading, Text } from "@stellar/design-system";
 
 import { Box } from "@/components/Box";
 import { ContentLayout } from "@/components/ContentLayout";
+import { Routes } from "@/config/settings";
+import { getSearchParams } from "@/helpers/getSearchParams";
 import { useStore } from "@/store/useStore";
 
 export const Success: FC = () => {
   const { t } = useTranslation();
   const { user } = useStore();
+  const navigate = useNavigate();
+  const searchParams = getSearchParams().toString();
+
+  // Redirect to /start if user is not set
+  useEffect(() => {
+    if (Object.keys(user).length === 0) {
+      navigate({ pathname: Routes.START, search: searchParams });
+    }
+  }, [user, navigate, searchParams]);
```

Comment:
We use the same check in a couple of places, so we could create this as a custom hook for more convenience.